libs/clj-jargon/src/clj_jargon/metadata.clj

```diff
@@ -42,41 +42,51 @@
       (.findMetadataValuesForCollection collection-ao dir-path)
       (.findMetadataValuesForDataObject data-ao dir-path))))
 
+(defn- get-metadata-by-query
+  [{^DataObjectAO data-ao :dataObjectAO
+    ^CollectionAO collection-ao :collectionAO
+    :as cm} path query]
+  (mapv avu2map
+    (if (is-dir? cm path)
+      (.findMetadataValuesByMetadataQueryForCollection collection-ao query path)
+      (.findMetadataValuesForDataObjectUsingAVUQuery data-ao query path))))
+
 (defn get-attribute
-  "Returns a list of avu maps for set of attributes associated with dir-path"
-  [cm dir-path attr]
+  "Returns a list of avu maps for a specific attribute associated with dir-path"
+  [{^DataObjectAO data-ao :dataObjectAO
+    ^CollectionAO collection-ao :collectionAO
+    :as cm} dir-path attr]
   (validate-path-lengths dir-path)
-  (filter
-    #(= (:attr %1) attr)
-    (get-metadata cm dir-path)))
-
-(defn get-attributes
```

Comment:
This function was entirely unused so I simply deleted it. If it's needed again, we can probably implement it with an `IN` query.